examples/pil-fun/src/Language/PilFun/Pretty.idr

```diff
@@ -19,72 +19,110 @@ import System.Random.Pure.StdGen
+ where
+  IsInfix : (a : Ty ** b : Ty ** to : MaybeTy ** funSig === ([<a, b] ==> to)) -> (isPure : Bool) -> IdrisCondition funSig True isPure
+  NotInfix : (isPure : Bool) -> IdrisCondition funSig False isPure
+
+public export
+data LuaCondition : FunSig -> (isInfix : Bool) -> (isPure : Bool) -> Type
+  where
+    TrivialLuaCondition : LuaCondition funSig isInfix isPure
+
+public export
+data LanguageToCondition : (l : SupportedLanguage) -> FunSig -> (isInfix : Bool) -> (isPure : Bool) -> Type
+ where
+  [search l]
+  Scala3Cond : ScalaCondition funSig isInfix isPure -> LanguageToCondition Scala3 funSig isInfix isPure
+  Idris2Cond : IdrisCondition funSig isInfix isPure -> LanguageToCondition Idris2 funSig isInfix isPure
+  Lua5_4Cond : LuaCondition funSig isInfix isPure -> LanguageToCondition Lua5_4 funSig isInfix isPure
+
+public export
+data UniqNames : (l : SupportedLanguage) -> (funs : Funs) -> (vars : Vars) -> Type
+public export
+data NameIsNew : (l : SupportedLanguage) -> (funs : Funs) -> (vars : Vars) -> UniqNames l funs vars -> String -> Type
+
+data UniqNames : (l : SupportedLanguage) -> (funs : Funs) -> (vars : Vars) -> Type where
+  [search funs vars]
+  Empty   : UniqNames l [<] [<]
+  JustNew : (ss : UniqNames l funs vars) => (s : String) -> (0 _ : NameIsNew l funs vars ss s) => UniqNames l funs vars
+  NewFun  : (ss : UniqNames l funs vars) => (s : String) -> (0 _ : NameIsNew l funs vars ss s) =>
+            {default False isInfix : Bool} -> {default False isPure : Bool} ->
+            (languageCondition : LanguageToCondition l fun isInfix isPure) =>
+            UniqNames l (funs:<fun) vars
+  NewVar  : (ss : UniqNames l funs vars) => (s : String) -> (0 _ : NameIsNew l funs vars ss s) => UniqNames l funs ((:<) vars var mut)
+
+data NameIsNew : (l : SupportedLanguage) -> (funs : Funs) -> (vars : Vars) -> UniqNames l funs vars -> String -> Type where
+  E : NameIsNew l [<] [<] Empty x
+  J : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l funs vars (JustNew @{ss} s @{sub}) x
+  F : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l (funs:<fun) vars (NewFun @{ss} {isInfix} {isPure} s @{sub} @{infixCond}) x
+  V : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l funs ((:<) vars var mut) (NewVar @{ss} s @{sub}) x
 
 public export
 interface NamesRestrictions where
   reservedKeywords : SortedSet String
 
-rawNewName : Fuel -> (Fuel -> Gen MaybeEmpty String) =>
-             (vars : Vars) -> (funs : Funs) -> (names : UniqNames funs vars) ->
-             Gen MaybeEmpty (s ** NameIsNew names s)
+rawNewName : Fuel -> (l : SupportedLanguage) -> (Fuel -> Gen MaybeEmpty String) =>
+             (funs : Funs) -> (vars : Vars) -> (names : UniqNames l funs vars) ->
+             Gen MaybeEmpty (s ** NameIsNew l funs vars names s)
 
 export
-genNewName : Fuel -> (Fuel -> Gen MaybeEmpty String) =>
+genNewName : {l : SupportedLanguage} -> Fuel -> Gen MaybeEmpty String ->
```

Comment:
Is it so nesessary to make existing `Fuel -> Gen MaybeEmpty String` argument to be not-autoimplicit? It's okay in case when every or almost every usage needs to set this explicitly, but it this the case? I see that in some places `alphaNames` are used, but it seems that in other cases auto-implicit still can be used. Am I wrong?

Additionally, please, put `l` after the fuel and auto-implicits to make all generators' signatures to look more or less uniformly

Reply:
Since we already have a pretty printer which essentially use two different name generators internally, so in order to avoid confusion of generators we should use explicit passing instead. The order of implicit arguments was fixed by 8c6e2aa4.